### ISSUE TITLE
Very long Python import names

### ISSUE THREAD

**Issue by cmclaughlin** [hashicorp/terraform-cdk#464]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The Python style guide recommends wrapping long lines at 80 characters.

It's not possible with these sort of imports:

```
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfiguration  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationInputFormatConfiguration  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationInputFormatConfigurationDeserializer  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationInputFormatConfigurationDeserializerHiveJsonSerDe  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationOutputFormatConfiguration  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationOutputFormatConfigurationSerializer  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationOutputFormatConfigurationSerializerParquetSerDe  # noqa: E501
from cdktf_cdktf_provider_aws import \
    KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationSchemaConfiguration  # noqa: E501
```

In this case, it seems stripping off the `KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfiguration` prefix could help. Or perhaps some sort of nesting, i.e. if these were part of a module it would be possible to wrap lines on the `.` character.
i.e. perhaps `KinesisFirehoseDeliveryStream.ExtendedS3ConfigurationDataFormatConversionConfiguration`

**Comment by skorfmann** [comment]
Ok, this looks way too long indeed.

> In this case, it seems stripping off the KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfiguration prefix could help.

This might lead to conflicts across resources in a provide due to the way we export the types.

> Or perhaps some sort of nesting, i.e. if these were part of a module it would be possible to wrap lines on the . character.
i.e. perhaps KinesisFirehoseDeliveryStream.ExtendedS3ConfigurationDataFormatConversionConfiguration

This seems like a viable option. jsii actually supports [submodules](https://github.com/aws/jsii/blob/377eae2058bc09e5350a17ab5e8fac627e1fe688/gh-pages/content/specification/2-type-system.md#submodules) out of the box and we were thinking about introducing namespaces in other contexts as well. 

Would this help? @cmclaughlin

**Comment by cmclaughlin** [comment]
Yep, that second option is ok... it's not ideal, but it's better than nothing.

**Comment by cmclaughlin** [comment]
Grouping like this is possible....

```
from cdktf_cdktf_provider_aws import (
     KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationSchemaConfiguration, 
     KinesisFirehoseDeliveryStreamExtendedS3ConfigurationDataFormatConversionConfigurationOutputFormatConfigurationSerializerParquetSerDe
)
```

**Comment by jsteinich** [comment]
> > Or perhaps some sort of nesting, i.e. if these were part of a module it would be possible to wrap lines on the . character.
> > i.e. perhaps KinesisFirehoseDeliveryStream.ExtendedS3ConfigurationDataFormatConversionConfiguration
> 
> This seems like a viable option. jsii actually supports [submodules](https://github.com/aws/jsii/blob/377eae2058bc09e5350a17ab5e8fac627e1fe688/gh-pages/content/specification/2-type-system.md#submodules) out of the box and we were thinking about introducing namespaces in other contexts as well.

I've thought about using nested types as another option which should have a similar result.

**Comment by skorfmann** [comment]
> I've thought about using nested types as another option which should have a similar result.

Could you provide a short example of what you mean?

**Comment by jsteinich** [comment]
> > I've thought about using nested types as another option which should have a similar result.
> 
> Could you provide a short example of what you mean?

I guess my knowledge of typescript was lacking here. I was thinking inner classes (like in Java/C#) existed, but looks like what I had seen was actually using namespaces. While that may work, looks like the [jsii team recommends submodules](https://github.com/aws/jsii/issues/283).
